craft_parts/overlays/overlay_manager.py

```diff
@@ -110,3 +117,122 @@ def mkdirs(self) -> None:
             self._project_info.overlay_work_dir,
         ]:
             overlay_dir.mkdir(parents=True, exist_ok=True)
+
+    def refresh_packages_list(self) -> None:
+        """Update the list of available packages in the overlay system."""
+        if not self._overlay_fs:
+            raise RuntimeError("overlay filesystem not mounted")
+
+        mount_dir = self._project_info.overlay_mount_dir
+        with contextlib.suppress(SystemExit), pychroot.Chroot(mount_dir):
+            packages.Repository.refresh_packages_list()
+
+    def download_packages(self, package_names: List[str]) -> None:
+        """Download packages and populate the overlay package cache.
+
+        :param package_names: The list of packages to download.
+        """
+        if not self._overlay_fs:
+            raise RuntimeError("overlay filesystem not mounted")
+
+        mount_dir = self._project_info.overlay_mount_dir
+        with contextlib.suppress(SystemExit), pychroot.Chroot(mount_dir):
+            packages.Repository.download_packages(package_names)
+
+    def install_packages(self, package_names: List[str]) -> None:
+        """Install packages on the overlay area using chroot.
+
+        :param package_names: The list of packages to install.
+        """
+        if not self._overlay_fs:
+            raise RuntimeError("overlay filesystem not mounted")
+
+        mount_dir = self._project_info.overlay_mount_dir
+        with contextlib.suppress(SystemExit), pychroot.Chroot(mount_dir):
+            packages.Repository.install_packages(package_names)
+            shutil.rmtree("/var/cache", ignore_errors=True)
+
+    def fix_resolv_conf(self) -> None:
+        """Work around problems with pychroot when resolv.conf a symlink."""
+        resolv = self._project_info.overlay_mount_dir / "etc" / "resolv.conf"
```

Comment:
Here you're not checking if the overlay fs was mounted...

Reply:
This method is only called by the `*Mount` context managers right after the filesystem is mounted. It is another workaround for bad pychroot behavior, and should go away when we replace pychroot.